sympy/functions/elementary/piecewise.py

```diff
@@ -192,16 +200,26 @@ def eval(cls, *_args):
                         for i in c.atoms(Relational):
                             ic = i.canonical
                             if ic.rhs in (S.Infinity, S.NegativeInfinity):
-                                if not _c.has(ic.rhs):
-                                    # don't accept introduction of
-                                    # new Relationals with +/-oo
-                                    reps[i] = S.true
+                                inf = ic.rhs
+                                if orig_c.xreplace({x: inf}) == True:
+                                    # it was a redundant inf
+                                    reps[i] = True
                                 elif ('=' not in ic.rel_op and
-                                        c.xreplace({x: i.rhs}) !=
-                                        _c.xreplace({x: i.rhs})):
+                                        orig_c.xreplace({x: inf}) !=
+                                        c.xreplace({x: inf})):
+                                    # hack around inability of Interval to
+                                    # contain +/-oo
+                                    # >>> And(x >= -oo, x <= 3).as_set()
+                                    # Interval(-oo, 3)
+                                    # >>> _.as_relational(x)
+                                    # (x <= 3) & (-oo < x)
                                     reps[i] = Relational(
                                         i.lhs, i.rhs, i.rel_op + '=')
```

Comment:
Is it possible to take a different approach and just not use `as_set` at all?

Also if we want to be able to have intervals with `oo` then that's fine but wouldn't it be better just to fix `Interval` rather than work around it here?

Reply:
> wouldn't it be better just to fix `Interval` rather than work around it here?

see [here](https://github.com/sympy/sympy/issues/22073#issuecomment-925377833)

Reply:
But recall that someone suggested just returning a Union when infinities are included. I am trying that.